### ISSUE TITLE
dredd fails to compile HTTP transactions when the response description contains non-alphabetical runes

### ISSUE THREAD

**Issue by glabra** [apiaryio/dredd#1685]
**Describe the bug**
When the path object:
- has header parameter
- and contains 2 or more non-default responses
- and at least one response's description has non-alphabetical runes (like `$`, `日本語`)

dredd fails with this message:
`Unable to compile HTTP transactions from API description document '/api/bundle.yml': Cannot assign to read only property 'parent' of object '#<MemberElement>': Cannot assign to read only property 'parent' of object '#<MemberElement>'`

**To Reproduce**
Run dredd with:
```sh
docker run -it -v $PWD:/api -w /api apiaryio/dredd:13.0.1 dredd ./bundle.yml http://localhost
```

`bundle.yml` is minimal file to reproduce the problem so far:
```yml bundle.yml
openapi: '3.0.0'
info:
  title: sample OpenAPIv3 file
  version: '0.0.1'
paths:
  /endpoint:
    parameters:
    - name: id
      in: header
      schema:
        type: string
      example: 'foo'
    get:
      summary: bar
      responses:
        '200':
          description: baz
          content:
            application/json:
              schema:
                type: string
        '400':
          description: contains non-alphabetical runes like 日本語
          content:
            application/json:
              schema:
                type: string
        default:
          description: foobar
          content:
            application/json:
              schema:
                type: string
```

**Expected behavior**
No error.

**What is in your `dredd.yml`?**
```yaml
<empty>
```

**What's your `dredd --version` output?**
```
dredd v13.0.1 (Linux 4.19.76-linuxkit; x64)
```

**Does `dredd --loglevel=debug` uncover something?**
No.

Full log:
```
$ docker run -it -v $PWD:/api -w /api apiaryio/dredd:13.0.1 dredd --loglevel=debug ./bundle.yml http://localhost
2020-03-05T07:11:18.097Z - debug: Loading configuration file: ./dredd.yml
2020-03-05T07:11:18.104Z - debug: Dredd version: 13.0.1
2020-03-05T07:11:18.105Z - debug: Node.js version: v12.16.1
2020-03-05T07:11:18.105Z - debug: Node.js environment: node=12.16.1, v8=7.8.279.23-node.31, uv=1.34.0, zlib=1.2.11, brotli=1.0.7, ares=1.15.0, modules=72, nghttp2=1.40.0, napi=5, llhttp=2.0.4, http_parser=2.9.3, openssl=1.1.1d, cldr=35.1, icu=64.2, tz=2019c, unicode=12.1
2020-03-05T07:11:18.105Z - debug: System version: Linux 4.19.76-linuxkit x64
2020-03-05T07:11:18.115Z - debug: npm version: [object Object]
2020-03-05T07:11:18.116Z - debug: Configuration: {"_":["./bundle.yml"],"color":true,"loglevel":"debug","l":"debug","dry-run":null,"y":null,"hookfiles":null,"f":null,"language":"nodejs","a":"nodejs","require":null,"server":"http://localhost","g":null,"server-wait":3,"init":false,"i":false,"custom":{"cwd":"/api","argv":["--loglevel=debug","./bundle.yml","http://localhost"]},"j":[],"names":false,"n":false,"only":[],"x":[],"reporter":[],"r":[],"output":[],"o":[],"header":[],"h":[],"sorted":false,"s":false,"user":null,"u":null,"inline-errors":false,"e":false,"details":false,"d":false,"method":[],"m":[],"path":["./bundle.yml","./bundle.yml"],"p":["./bundle.yml","./bundle.yml"],"hooks-worker-timeout":5000,"hooks-worker-connect-timeout":1500,"hooks-worker-connect-retry":500,"hooks-worker-after-connect-wait":100,"hooks-worker-term-timeout":5000,"hooks-worker-term-retry":500,"hooks-worker-handler-host":"127.0.0.1","hooks-worker-handler-port":61321,"config":"./dredd.yml","$0":"/usr/local/bin/node /node_modules/.bin/dredd"}
2020-03-05T07:11:18.127Z - debug: No backend server process specified, starting testing at once
2020-03-05T07:11:18.128Z - debug: Running Dredd instance.
2020-03-05T07:11:18.129Z - debug: Resolving --require
2020-03-05T07:11:18.129Z - debug: Configuring reporters
2020-03-05T07:11:18.130Z - debug: Using 'base' reporter.
2020-03-05T07:11:18.131Z - debug: Configuring reporters: []
2020-03-05T07:11:18.131Z - debug: Using 'cli' reporter.
2020-03-05T07:11:18.131Z - debug: Preparing API description documents
2020-03-05T07:11:18.131Z - debug: Resolving locations of API description documents
2020-03-05T07:11:18.143Z - debug: Reading API description documents
2020-03-05T07:11:18.159Z - debug: Parsing API description documents
2020-03-05T07:11:18.233Z - debug: Compiling HTTP transactions from API description documents
2020-03-05T07:11:18.235Z - debug: Dredd instance run finished.
2020-03-05T07:11:18.235Z - error: Unable to compile HTTP transactions from API description document '/api/bundle.yml': Cannot assign to read only property 'parent' of object '#<MemberElement>': Cannot assign to read only property 'parent' of object '#<MemberElement>'
2020-03-05T07:11:18.235Z - debug: Using native process.exit() method to terminate the Dredd process with status '1'.
2020-03-05T07:11:18.236Z - debug: No backend server process to terminate.
$
```

**Can you send us failing test in a Pull Request?**
No, but if you ask I'd glad to do so.

**Comment by kylef** [comment]
Hi @glabra, thanks for giving the detailed report. I've had a quick look and the bug is in our parser (fury-adapter-oas3-parser). I'll mention for anyone else digging into this, the bug is unrelated to to the non-alphabetical characters in description, just the presence of a header when you have multiple responses.

We'll get this fixed soon.

**Comment by kylef** [comment]
Concensed OAS 3 document to reproduce:

```yaml
openapi: '3.0.0'
info:
  title: sample OpenAPIv3 file
  version: '0.0.1'
paths:
  /endpoint:
    parameters:
    - name: id
      in: header
      example: 'foo'
    get:
      responses:
        '200':
          description: a
        '400':
          description: b
```

**Comment by kylef** [comment]
Fix has been proposed in https://github.com/apiaryio/api-elements.js/pull/417

**Comment by kylef** [comment]
This is fixed in [Dredd 13.0.2](https://github.com/apiaryio/dredd/releases/tag/dredd%4013.0.2)
